src/log.rs

```diff
@@ -425,3 +425,66 @@ where
         self.next()
     }
 }
+
+/// Returns an iterator over entries in the log stored in a thread-local variable.
+pub fn iter_thread_local<T, I, D>(
+    local_key: &'static std::thread::LocalKey<std::cell::RefCell<Log<T, I, D>>>,
+) -> ThreadLocalRefIterator<T, I, D>
+where
+    T: Storable,
+    I: Memory,
+    D: Memory,
+{
+    ThreadLocalRefIterator {
+        log: local_key,
+        buf: vec![],
+        pos: 0,
+    }
+}
+
+pub struct ThreadLocalRefIterator<T, I, D>
+where
+    T: Storable + 'static,
+    I: Memory + 'static,
+    D: Memory + 'static,
+{
+    log: &'static std::thread::LocalKey<std::cell::RefCell<Log<T, I, D>>>,
+    buf: Vec<u8>,
+    pos: u64,
+}
+
+impl<T, I, D> Iterator for ThreadLocalRefIterator<T, I, D>
+where
+    T: Storable,
+    I: Memory,
+    D: Memory,
+{
+    type Item = T;
+
+    fn next(&mut self) -> Option<T> {
+        self.log.with(
+            |log| match log.borrow().read_entry(self.pos, &mut self.buf) {
+                Ok(()) => {
+                    self.pos = self.pos.saturating_add(1);
+                    Some(T::from_bytes(Cow::Borrowed(&self.buf)))
+                }
+                Err(NoSuchEntry) => None,
+            },
+        )
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        let count = self.log.with(|cell| cell.borrow().len());
```

Comment:
I suppose this is the same as in count(), so can be simplified

```rs
fn size_hint(&self) -> (usize, Option<usize>) {
  (self.count(), None)
}
```

Reply:
Unfortunately, that's impossible because `count()` consumes the iterator. I implemented it the other way around.